examples/multi-threaded/main.c

```diff
@@ -61,14 +63,18 @@ static void pcb(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
 static void fn(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
   if (ev == MG_EV_HTTP_MSG) {
     struct mg_http_message *hm = (struct mg_http_message *) ev_data;
+      char param[64];
```

Comment:
Why not passing the whole body, which could be a JSON message?
Like, 

```c
struct mg_str data = hm->body;
if (data.len > 512) data = mg_str("{}");
...
mg_http_reply(...., "%.*s\n", (int) data.len, data.ptr);
```

Reply:
The OP question was how to access `ev_data` to be able to get the headers, that's why I thought on something that was simply adding `?param=something` to the URL. Yes, passing the body can be easily shown with curl, will do that.